tests/ssg_test_suite/rule.py

```diff
@@ -344,16 +363,45 @@ def _check_and_record_rule_scenario(self, scenario, remote_rule_dir, rule_id, re
         self._current_result.scenario = common.Scenario_run(rule_id, scenario.script)
         self._current_result.when = self.test_timestamp_str
 
-        self._check_rule_scenario(scenario, remote_rule_dir, rule_id, remediation_available)
+        with self.copy_of_datastream():
+            self._check_rule_scenario(scenario, remote_rule_dir, rule_id, remediation_available)
         self.results.append(self._current_result.save_to_dict())
 
+    @contextlib.contextmanager
+    def copy_of_datastream(self, new_filename=None):
+        old_filename = self.datastream
+        if not new_filename:
+            _, new_filename = tempfile.mkstemp(prefix="ssgts_ds_modified", dir="/tmp")
+        shutil.copy(old_filename, new_filename)
+        self.datastream = new_filename
+        yield new_filename
+        self.datastream = old_filename
+        os.unlink(new_filename)
+
+    def _change_variable_value(self, varname, value):
+        XSLT_FILENAME = "/tmp/change.xslt"
```

Comment:
This would be better as a temporary file instead

Reply:
OK